### ISSUE TITLE
Dynamic styles error:  not implemented: EvaluateResultValue::Callback

### ISSUE THREAD

**Issue by BjornHMS** [Dwlad90/stylex-swc-plugin#49]
Is @stylexjs/babel-plugin required? 
I tried running the stylex-swc-plugin/apps/[nextjs-example](https://github.com/Dwlad90/stylex-swc-plugin/tree/master/apps/nextjs-example) project.
When I try to run it I get this error "Error: Cannot find module '**@stylexjs/babel-plugin**'"
**@stylexjs/babel-plugin** not included in package.json

I had the same problem when I set up a different project to use stylex-swc-plugin, that time I just installed **@stylexjs/babel-plugin** and kept going. It worked, but when I tried using a dynamic styles it would crash.

Have I perhaps configured it wrong and it's still trying to use babel instead of swc?
Or is this intended and **@stylexjs/babel-plugin** was just missing from the package.json

**Comment by BjornHMS** [comment]
> When you use [stylex-swc-plugin](https://github.com/Dwlad90/stylex-swc-plugin) in nextjs, you no longer need babel-plugin. You can try the boilerplate here: https://github.com/Dwlad90/nextjs-app-dir-stylex

That actually has **@stylexjs/babel-plugin** though.. so I do need it?

**Comment by Dwlad90** [comment]
Hey @BjornHMS,
Thank you so much for finding the bugs.

The problem with the babel plugin occurs because it is not added to the dependencies of the `@stylexswc/nextjs-plugin`.
The plugin itself is not used, it is needed in the dependencies, since the function for processing generated styles is in this package.

https://github.com/Dwlad90/stylex-swc-plugin/blob/2bdc0d699d8cdb7f56be24452451f6367ff79331/packages/nextjs-plugin/src/custom-webpack-plugin.ts#L89

As for dynamic styles, this is indeed a bug of the SWC transformer, which I will try to fix as soon as possible.

**Comment by BjornHMS** [comment]
@Dwlad90
You're very welcome.
And thanks for your work on this package.

In the mean time, I will get by doing dynamic styles like this:
```typescript
  const dynamicStyles = useMemo(() => {
    const { className, style = {} } = { ...stylex.props(styles.tag_input) };
    (style as CSSProperties).fontSize = `${8 * count}px`;
    return { className, style };
  }, [count]);
```

**Comment by Dwlad90** [comment]
@BjornHMS,

Release candidate `0.3.1-rc.1` with a fix for this issue has been uploaded to NPM. Could you please check if the issue is actually fixed?

**Comment by BjornHMS** [comment]
> Release candidate `0.3.1-rc.1` with a fix for this issue has been uploaded to NPM. Could you please check if the issue is actually fixed?

Tested, and the issue has been resolved.
Thank you very much @Dwlad90 for the swift fix.

**Comment by BjornHMS** [comment]
@Dwlad90 I reopened, because It's crashing if the function has a condition.

For example this works
```typescript
const styles = stylex.create({
  fontSize: (size: number) => ({ fontSize: `${8 * size}px` })
});
```

But this does not.

```typescript
const styles = stylex.create({
  fontSize: (size: number) => ({ fontSize: size < 10 ? '1em' : '2em' })
});
```
neither does this

```typescript
const styles = stylex.create({
  fontSize: (size?: number) => ({ fontSize: size ?? '1em' })
});
```

```
The application panicked (crashed).
Message:  Test of condition must be an expression
Location: crates\stylex-shared\src\shared\utils\js\evaluate.rs:326

Run with COLORBT_SHOW_HIDDEN=1 environment variable to disable frame filtering.
━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ BACKTRACE ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━
 1: napi_register_module_v1
    at <unknown source file>
 2: napi_register_module_v1
    at <unknown source file>
 3: napi_register_module_v1
    at <unknown source file>
 4: napi_register_module_v1
    at <unknown source file>
 5: napi_register_module_v1
    at <unknown source file>
 6: napi_register_module_v1
    at <unknown source file>
 7: napi_register_module_v1
    at <unknown source file>
 8: napi_register_module_v1
    at <unknown source file>
 9: napi_register_module_v1
    at <unknown source file>
10: napi_register_module_v1
    at <unknown source file>
11: napi_register_module_v1
    at <unknown source file>
12: napi_register_module_v1
    at <unknown source file>
13: v8::base::CPU::has_idiva
    at <unknown source file>
14: v8::SharedValueConveyor::SharedValueConveyor
    at <unknown source file>
15: v8::SharedValueConveyor::SharedValueConveyor
    at <unknown source file>
16: v8::SharedValueConveyor::SharedValueConveyor
    at <unknown source file>
17: v8::SharedValueConveyor::SharedValueConveyor
    at <unknown source file>
18: v8::PropertyDescriptor::writable
    at <unknown source file>
19: v8::PropertyDescriptor::writable
    at <unknown source file>
20: v8::PropertyDescriptor::writable
    at <unknown source file>
21: v8::PropertyDescriptor::writable
    at <unknown source file>
22: v8::PrimitiveArray::Length
    at <unknown source file>
23: v8::PrimitiveArray::Length
    at <unknown source file>
24: v8::PrimitiveArray::Length
    at <unknown source file>
25: v8::base::CPU::has_sse41
    at <unknown source file>
26: v8::base::CPU::has_sse41
    at <unknown source file>
27: node::CallbackScope::~CallbackScope
    at <unknown source file>
28: node::CallbackScope::~CallbackScope
    at <unknown source file>
29: node_module_register
    at <unknown source file>
30: uv_timer_stop
    at <unknown source file>
31: uv_timer_stop
    at <unknown source file>
32: uv_update_time
    at <unknown source file>
33: uv_run
    at <unknown source file>
34: node::SpinEventLoop
    at <unknown source file>
35: X509_STORE_get_cleanup
    at <unknown source file>
36: node::Start
    at <unknown source file>
37: node::Start
    at <unknown source file>
38: AES_cbc_encrypt
    at <unknown source file>
39: inflateValidate
    at <unknown source file>
40: BaseThreadInitThunk
    at <unknown source file>
41: RtlUserThreadStart
    at <unknown source file>
```

I'm not sure if this is a swc problem or a styleX problem.

**Comment by Dwlad90** [comment]
Thanks @BjornHMS for finding similar issues. This was a problem with the SWC transformer.

The fix is ​​available in the "0.3.1-rc.2" plugin version

**Comment by BjornHMS** [comment]
> The fix is ​​available in the "0.3.1-rc.2" plugin version

You've fixed it! Thank you very much @Dwlad90
I will continue to submit issue's if I encounter them ;)
